test/input-states-mocha.js

```diff
@@ -0,0 +1,333 @@
+                }
+            });
+        });
+        return test.createNetwork({
+            upstream1: {
+                getComponent: componentFactory({
+                    outPorts:{
+                        output:{}
+                    }
+                })
+            },
+            upstream2: {
+                getComponent: componentFactory({
+                    outPorts:{
+                        output:{}
+                    }
+                })
+            },
+            test: {
+                getComponent: getComponent
+            }
+        }).then(function(network){
+            network.graph.addEdge('upstream1', 'output', 'test', 'input');
+            network.graph.addEdge('upstream1', 'output', 'test', 'inputa');
+            network.graph.addEdge('upstream2', 'output', 'test', 'input');
+            network.graph.addEdge('upstream2', 'output', 'test', 'inputa');
+            return network;
+        }).then(function(nw){
+            network = nw;
+        });
+    });
+    afterEach(function(){
+        network.stop();
+    });
+    it('should exist as a function', function() {
+        node.vni().inputStates.should.be.a('function');
+    });
+    it('should fail if vnid is not a string', function() {
+        expect(function(){return node.vni(-1).inputStates('input');}).to.throw(Error);
+    });
+    it('should fail if invalid port name', function() {
+        expect(function(){return node.vni().inputStates('bogus');}).to.throw(Error);
+    });
+    it('should fail if not addressable', function() {
+        expect(function(){return node.vni().inputStates('input', 0);}).to.throw(Error);
+        expect(function(){return node.vni().inputStates('input', 0, {});}).to.throw(Error);
+    });
+    it('should return undefined when no state exists', function() {
+        expect(node.vni().inputStates('input') ).to.be.undefined;
+        expect(node.vni().inputStates('inputa', 0) ).to.be.undefined;
+    });
+    it('should list all ports', function() {
+        node.vni().inputStates().should.have.all.keys(_.keys(node.inPorts));
+    });
+    it('should return an undefined state when an optional port that is not attached to anything', function() {
+        expect(node.vni().inputStates('input2')).to.be.undefined;
+    });
+    it('should return [] when a multi port with nothing attached', function() {
+        expect(node.vni().inputStates('inputb')).to.eql([]);
+    });
+    it('should set and get a new IIP input state', function() {
+        var state = { vnid: '001',
+                   data: 'hello world',
+                   lm: createLm() }
+
+        var context = node.vni().inputStates({'input': state});
+        context.should.exist;
+
+        var result = node.vni().inputStates('input'); 
+        verifyInputState( result, state );
+    });
+    it('should treat undefined socketIndex the same as without it', function() {
+        var state = { vnid: '001',
+                   data: 'hello world',
```

Comment:
inconsistent alignment on the data and lm elements here and multiple other places with declaring state values